src/libraries/System.Private.CoreLib/src/System/Globalization/DateTimeFormat.cs

```diff
@@ -1112,6 +1112,235 @@ private static StringBuilder FormatStringBuilder(DateTime dateTime, ReadOnlySpan
             return FormatCustomized(dateTime, format, dtfi, offset, result: null);
         }
 
+        internal static bool IsValidCustomDateFormat(ReadOnlySpan<char> format, bool throwOnError)
+        {
+            int length = format.Length;
+            int i = 0;
+
+            while (i < length)
+            {
+                switch (format[i])
+                {
+                    case '\\':
+                        if (i == length - 1)
+                        {
+                            if (throwOnError)
+                            {
+                                throw new FormatException(SR.Format_InvalidString);
```

Comment:
Is this hot and worth a Throw Helper?

Reply:
This is rare to happen so I don't think it is worth optimizing it.